backend/build_docs.py

```diff
@@ -16,134 +17,120 @@
-                copied_files[src_path] = (
-                    f"OpenROAD Manpages - {src_path.split('/')[-1]}"
-                )
-            elif 'yosys' in src:
-                copied_files[src_path] = src_path.split('/data/rtdocs/')[-1]
-            elif 'OpenSTA' in src:
-                copied_files[src_path] = opensta_docs_url
-            else:
-                copied_files[src_path] = src_path
-
-    return copied_files
+            update_src(src_path, src_path)
 
 
-def copy_file_track_src(src: str, dst: str) -> dict[str, str]:
-    copied_files = {}
-
+def copy_file_track_src(src: str, dst: str) -> None:
     if not os.path.exists(src):
         logging.debug(f'File {src} does not exist. Exiting.')
         sys.exit(1)
 
     if os.path.isfile(src):
+        if os.path.exists(dst):
+            base, ext = os.path.splitext(dst)
+            counter = 2
+            while os.path.exists(dst):
+                new_file_name = f'{base}_{counter}{ext}'
+                logging.debug(f'File {dst} already exists. Renaming to {new_file_name}')
+                dst = new_file_name
+                counter += 1
+
         shutil.copy2(src, dst)
 
         dst_path = dst.split('backend/')[-1]
 
-        if 'OR_docs' in dst:
-            copied_files[dst_path] = (
-                f"{or_docs_url}/{src.split('_sources/')[-1].replace('.md', '.html')}"
-            )
-        elif 'ORFS_docs' in dst:
-            copied_files[dst_path] = (
-                f"{orfs_docs_url}/{src.split('_sources/')[-1].replace('.md', '.html')}"
-            )
-        elif 'manpages' in dst:
-            copied_files[dst_path] = (
-                f"OpenROAD Manpages - {src.split('data/markdown/')[-1]}"
-            )
-        elif 'yosys' in dst:
-            copied_files[dst_path] = src.split('/data/rtdocs/')[-1]
-        elif 'OpenSTA' in dst:
-            copied_files[dst_path] = opensta_docs_url
-        else:
-            copied_files[dst_path] = dst_path
-
-    return copied_files
-
+        update_src(src, dst_path)
 
-def copy_tree_track_src(src: str, dst: str) -> dict[str, str]:
-    copied_files = {}
 
+def copy_tree_track_src(src: str, dst: str) -> None:
     if not os.path.exists(src):
         logging.debug(f'Folder {src} does not exist. Exiting.')
         sys.exit(1)
 
     for root, _, files in os.walk(src):
         rel_path = os.path.relpath(root, src)
-        if rel_path == '.':
-            dst_dir = dst
-        else:
-            dst_dir = os.path.join(dst, rel_path)
+        dst_dir = dst if rel_path == '.' else os.path.join(dst, rel_path)
         os.makedirs(dst_dir, exist_ok=True)
 
         for file in files:
             src_file = os.path.join(root, file)
             dst_file = os.path.join(dst_dir, file)
-            shutil.copy2(src_file, dst_file)
 
-            dst_path = dst_file.split('backend/')[-1]
+            if os.path.exists(dst_file):
+                base, ext = os.path.splitext(file)
+                counter = 2
```

Comment:
why does the counter start from 2?

Reply:
The 2nd file with the same name will be renamed `file_2.md` and so on.